service/worker/scheduler/spec.go

```diff
@@ -34,30 +34,28 @@ import (
 	"go.temporal.io/server/common/primitives/timestamp"
 )
 
-const (
-	defaultJitter = 1 * time.Second
-)
-
 type (
 	compiledSpec struct {
-		spec     *schedpb.ScheduleSpec
-		tz       *time.Location
-		calendar []*compiledCalendar
-		excludes []*compiledCalendar
+		spec          *schedpb.ScheduleSpec
+		tz            *time.Location
+		calendar      []*compiledCalendar
+		excludes      []*compiledCalendar
+		defaultJitter time.Duration
 	}
 )
 
-func newCompiledSpec(spec *schedpb.ScheduleSpec) (*compiledSpec, error) {
+func newCompiledSpec(spec *schedpb.ScheduleSpec, defaultJitter time.Duration) (*compiledSpec, error) {
 	tz, err := loadTimezone(spec)
 	if err != nil {
 		return nil, err
 	}
 
 	cspec := &compiledSpec{
-		spec:     spec,
-		tz:       tz,
-		calendar: make([]*compiledCalendar, len(spec.Calendar)),
-		excludes: make([]*compiledCalendar, len(spec.ExcludeCalendar)),
+		spec:          spec,
+		tz:            tz,
+		calendar:      make([]*compiledCalendar, len(spec.Calendar)),
+		excludes:      make([]*compiledCalendar, len(spec.ExcludeCalendar)),
+		defaultJitter: defaultJitter,
```

Comment:
I think we should just have default jitter set to 0 and make it optional. If user do not supply one, we use 0 jitter.

Reply:
I'm suggesting to remove the default entirely. Including update the [proto comment ](https://github.com/temporalio/api/blob/f4d9850a210f4462467c3344062a267ed294eb29/temporal/api/schedule/v1/message.proto#L114) to say default is 0. We would only use jitter if user ask to do so. I see no value of using a 1s jitter as default.